compaction/task_manager_module.hh

```diff
@@ -149,6 +149,70 @@ protected:
     virtual future<> run() override;
 };
 
+class offstrategy_compaction_task_impl : public compaction_task_impl {
+public:
+    offstrategy_compaction_task_impl(tasks::task_manager::module_ptr module,
+            tasks::task_id id,
+            unsigned sequence_number,
+            std::string keyspace,
+            std::string table,
+            std::string entity,
+            tasks::task_id parent_id) noexcept
+        : compaction_task_impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(entity), parent_id)
+    {
+        // FIXME: add progress units
+    }
+
+    virtual std::string type() const override {
+        return "offstrategy compaction";
+    }
+protected:
+    virtual future<> run() override = 0;
+};
+
+class offstrategy_keyspace_compaction_task_impl : public offstrategy_compaction_task_impl {
+private:
+    sharded<replica::database>& _db;
+    std::vector<table_id> _table_infos;
+    bool& _res;
+public:
+    offstrategy_keyspace_compaction_task_impl(tasks::task_manager::module_ptr module,
+            std::string keyspace,
+            sharded<replica::database>& db,
+            std::vector<table_id> table_infos,
+            bool& res) noexcept
+        : offstrategy_compaction_task_impl(module, tasks::task_id::create_random_id(), module->new_sequence_number(), std::move(keyspace), "", "", tasks::task_id::create_null_id())
+        , _db(db)
+        , _table_infos(std::move(table_infos))
+        , _res(res)
+    {}
+protected:
+    virtual future<> run() override;
+};
+
+class shard_offstrategy_keyspace_compaction_task_impl : public offstrategy_compaction_task_impl {
+private:
```

Comment:
shard means shard local right? perhaps local would have been a better prefix. it's not immediately clear to me what shard means, unless  i peek into the implementation.